Source/WebKit/WebProcess/Cache/WebCacheStorageConnection.cpp

```diff
@@ -87,7 +96,10 @@ void WebCacheStorageConnection::batchDeleteOperation(WebCore::DOMCacheIdentifier
 
 void WebCacheStorageConnection::batchPutOperation(WebCore::DOMCacheIdentifier cacheIdentifier, Vector<Record>&& records, WebCore::DOMCacheEngine::RecordIdentifiersCallback&& callback)
 {
-    connection().sendWithAsyncReply(Messages::NetworkStorageManager::CacheStoragePutRecords(cacheIdentifier, records), WTFMove(callback));
+    auto crossThreadRecords = WTF::map(records, [](auto&& record) {
+        return toCrossThreadRecord(WTFMove(record));
```

Comment:
We could also update this code to directly pass cross thread records to this method.